src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CoseKeyHelpers.cs

```diff
@@ -0,0 +1,261 @@
+
+                ReadCoseKeyLabel(reader, CoseKeyLabel.EcCrv);
+                CoseCrvId crv = (CoseCrvId)reader.ReadInt32();
+
+                if (!IsValidKtyCrvCombination(kty, crv))
+                {
+                    throw new FormatException("Invalid kty/crv combination in COSE key.");
+                }
+
+                ecParams.Curve = MapCoseCrvToECCurve(crv);
+
+                ReadCoseKeyLabel(reader, CoseKeyLabel.EcX);
+                ecParams.Q.X = reader.ReadByteString();
+
+                ReadCoseKeyLabel(reader, CoseKeyLabel.EcY);
+                ecParams.Q.Y = reader.ReadByteString();
+
+                reader.ReadEndMap();
+
+                return (ecParams, algName);
+            }
+            catch (InvalidOperationException e)
+            {
+                throw new FormatException("Invalid COSE_key format in CBOR document", e);
+            }
+
+            static bool IsValidKtyCrvCombination(CoseKeyType kty, CoseCrvId crv)
+            {
+                return (kty, crv) switch
+                {
+                    (CoseKeyType.EC2, CoseCrvId.P256 or CoseCrvId.P384 or CoseCrvId.P521) => true,
+                    (CoseKeyType.OKP, CoseCrvId.X255519 or CoseCrvId.X448 or CoseCrvId.Ed25519 or CoseCrvId.Ed448) => true,
+                    _ => false,
+                };
+                ;
+            }
+
+            static ECCurve MapCoseCrvToECCurve(CoseCrvId crv)
+            {
+                return crv switch
+                {
+                    CoseCrvId.P256 => ECCurve.NamedCurves.nistP256,
+                    CoseCrvId.P384 => ECCurve.NamedCurves.nistP384,
+                    CoseCrvId.P521 => ECCurve.NamedCurves.nistP521,
+                    CoseCrvId.X255519 or
+                    CoseCrvId.X448 or
+                    CoseCrvId.Ed25519 or
+                    CoseCrvId.Ed448 => throw new NotImplementedException("OKP type curves not implemented."),
+                    _ => throw new FormatException("Unrecognized COSE crv value."),
+                };
+            }
+
+            static HashAlgorithmName MapCoseKeyAlgToHashAlgorithmName(CoseKeyAlgorithm alg)
+            {
+                return alg switch
+                {
+                    CoseKeyAlgorithm.ES256 => HashAlgorithmName.SHA256,
+                    CoseKeyAlgorithm.ES384 => HashAlgorithmName.SHA384,
+                    CoseKeyAlgorithm.ES512 => HashAlgorithmName.SHA512,
+                    _ => throw new FormatException("Unrecognized COSE alg value."),
+                };
+            }
+        }
+
+        private static void WriteCoseKeyLabel(CborWriter writer, CoseKeyLabel label)
+        {
+            writer.WriteInt32((int)label);
+        }
+
+        private static CoseKeyLabel ReadCoseKeyLabel(CborReader reader, CoseKeyLabel? expectedLabel = null)
+        {
+            var label = (CoseKeyLabel)reader.ReadInt32();
+            if (label != expectedLabel)
```

Comment:
should this be `expectedLabel != null && ...`

Reply:
If expectedLabel is null then the existing condition will always be false.